### PR TITLE
Add invertSelf to the DOMMatrix object

### PR DESCRIPTION
**Pull request by Pomax**
This should address https://github.com/Automattic/node-canvas/issues/1648 by adding the implementation for `invertSelf()`.

- [x] added tests for `inverse()` and `invertSelf()`
- [x] added the change to the unreleased `added` section of the Changelog

### CODE REVIEW

**Comment by Pomax** [comment]
I have no idea what Mocha/Node wants me to do here to make the tests pass. I've used the exact same code and style as used in the rest of the tests, and yet it complains about a syntax error, where there literally are no syntax errors.

File: test/dommatrix.test.js [hunk 1]
```diff
@@ -403,8 +403,51 @@ describe('DOMMatrix', function () {
...
+      assertApprox(d.m44, -0.6610169491525424)
+    })
+  })
+
+  describe('inverse', function () {
+    it('preserves the original DOMMatrix', function() {
+      var d = new DOMMatrix(Avals)
+      d.inverse()
+      assert.strictEqual(d.m11, Avals[0])
+      assert.strictEqual(d.​m12, Avals[1])
```
Comment (Pomax) [inline test/dommatrix.test.js:434]: This makes no sense whatsoever:
```
node-canvas\test\dommatrix.test.js:434
      assert.strictEqual(d.​m12, Avals[1])


SyntaxError: Invalid or unexpected token
    at wrapSafe (internal/modules/cjs/loader.js:1167:16)
    at Module._compile (internal/modules/cjs/loader.js:1215:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1272:10)
    at Module.load (internal/modules/cjs/loader.js:1100:32)
    at Function.Module._load (internal/modules/cjs/loader.js:962:14)
    at Module.require (internal/modules/cjs/loader.js:1140:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at C:\Users\Mike\Documents\Git\temp\node-canvas\node_modules\mocha\lib\mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (C:\Users\Mike\Documents\Git\temp\node-canvas\node_modules\mocha\lib\mocha.js:247:14)
    at Mocha.run (C:\Users\Mike\Documents\Git\temp\node-canvas\node_modules\mocha\lib\mocha.js:576:10)
    at Object.<anonymous> (C:\Users\Mike\Documents\Git\temp\node-canvas\node_modules\mocha\bin\_mocha:637:18)
    at Module._compile (internal/modules/cjs/loader.js:1251:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1272:10)
    at Module.load (internal/modules/cjs/loader.js:1100:32)
    at Function.Module._load (internal/modules/cjs/loader.js:962:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

Comment (Pomax) [reply test/dommatrix.test.js:434]: @chearon I don't supposed you'd know how to fix this? =S

Comment (chearon) [reply test/dommatrix.test.js:434]: There's a bunch of zero-width spaces in there that show up when I use Vim:

Screen Shot 2020-08-14 at 11 19 06 AM

File: lib/DOMMatrix.js [hunk 2]
```diff
@@ -445,9 +445,132 @@ DOMMatrix.prototype.flipY = function () {
...
+
+  var det = m[0] * inv[0] + m[1] * inv[4] + m[2] * inv[8] + m[3] * inv[12];
+
+  if (det === 0) {
+    this._values = m.map(v => NaN);
+    this._is2D = false;
+    return;
+  }
+  
+  this._values = inv.map(v => v/det);
```
Comment (zbjornson) [inline lib/DOMMatrix.js:573]: Instead of making a copy/spare array here, could you divide `inv` in place?

File: test/dommatrix.test.js (hunk 1 above)
Comment (Pomax) [reply test/dommatrix.test.js:434]: how in the world... I copied the lines from the other tests O_o

Alright, let's clean that up I guess, good catch.

File: lib/DOMMatrix.js (hunk 2 above)
Comment (Pomax) [reply lib/DOMMatrix.js:573]: good call.

**Comment by Pomax** [comment]
PR updated.

**Comment by Pomax** [comment]
It looks like the `inverse()` function isn't returning anything, so I'm having a look at why that might be - I thought I was using the same code as other functions that return copy-constructed matrices.

File: lib/DOMMatrix.js [hunk 3]
```diff
@@ -446,8 +446,132 @@ DOMMatrix.prototype.inverse = function () {
...
+          m[8] * m[3] * m[5];
+
+  inv[15] = m[0] * m[5] * m[10] -
+          m[0] * m[6] * m[9] -
+          m[4] * m[1] * m[10] +
+          m[4] * m[2] * m[9] +
+          m[8] * m[1] * m[6] -
+          m[8] * m[2] * m[5];
+
+  var det = m[0] * inv[0] + m[1] * inv[4] + m[2] * inv[8] + m[3] * inv[12];
```
Comment (Pomax) [inline lib/DOMMatrix.js:564]: hm, given that this relies on only `inv` values 0, 4, 8, and 12, we can probably shortcut this operation by computing those first, checking the determinant, and returning early. 

I also just realised there is no test for 4x4 that can't be inverted, so I should probably add one before this gets merged.

**Comment by Pomax** [comment]
@zbjornson? =)

**Pull request merged**